src/check-update.ts

```diff
@@ -0,0 +1,250 @@
+
+const DENO_SLACK_SDK = "deno_slack_sdk";
+const DENO_SLACK_API = "deno_slack_api";
+const DENO_SLACK_HOOKS = HOOKS_TAG.substring(0, HOOKS_TAG.indexOf("@"));
+const DENO_SLACK_BUILDER = BUILDER_TAG.substring(0, BUILDER_TAG.indexOf("@"));
+const DENO_SLACK_RUNTIME = RUNTIME_TAG.substring(0, RUNTIME_TAG.indexOf("@"));
+const IMPORT_MAP_SDKS = [DENO_SLACK_SDK, DENO_SLACK_API];
+const SLACK_JSON_SDKS = [
+  DENO_SLACK_HOOKS, // should be the only one needed once the get-scripts hook is supported
+  DENO_SLACK_BUILDER, // but just in case, we can look for builder and runtime as well!
+  DENO_SLACK_RUNTIME,
+];
+
+interface UpdateResponse {
+  update: boolean;
+  breaking: boolean;
+  message: string;
+  error: {
+    message: string;
+  };
+}
+
+interface VersionMap {
+  [key: string]: {
+    current: string;
+    latest: string;
+    update: boolean;
+    breaking: boolean;
+    error: string;
+  };
+}
+
+export const checkForSDKUpdates = async () => {
+  const versionMap: VersionMap = await createVersionMap();
+  const updateResp = createUpdateResp(versionMap);
+  return updateResp;
+};
+
+/**
+ * createVersionMap creates an object that contains each dependency,
+ * featuring information about the current and latest versions, as well
+ * as if breaking changes are present and if any errors occurred during
+ * version retrieval.
+ */
+async function createVersionMap() {
+  const versionMap: VersionMap = await readProjectDependencies();
+
+  // Check each dependency for updates, classify update as breaking or not,
+  // craft message with information retrieved, and note any error that occurred.
+  for (const [sdk, value] of Object.entries(versionMap)) {
+    if (value) {
+      const current = versionMap[sdk].current;
+      let latest = "", error = null;
+
+      try {
+        latest = await fetchLatestModuleVersion(sdk);
+      } catch (err) {
+        error = err;
+      }
+
+      const update = current !== latest;
+      const breaking = hasBreakingChange(current, latest);
+
+      versionMap[sdk] = {
+        ...versionMap[sdk],
+        latest,
+        update,
+        breaking,
+        error,
+      };
+    }
+  }
+
+  return versionMap;
+}
+
+/** readProjectDependencies reads from possible dependency files
+ * (import_map.json, slack.json) and maps them to the versionMap
+ * containing each dependency's update information
+ */
+async function readProjectDependencies(): Promise<VersionMap> {
+  const cwd = Deno.cwd();
+  const versionMap: VersionMap = {};
+
+  // Find SDK component versions in import map, if available
+  const map = await getJson(`${cwd}/import_map.json`);
+  for (const sdkUrl of Object.values(map.imports) as string[]) {
+    for (const sdk of IMPORT_MAP_SDKS) {
+      if (sdkUrl.includes(sdk)) {
+        versionMap[sdk] = {
+          latest: "",
+          update: false,
+          breaking: false,
+          error: "",
+          current: extractVersion(sdkUrl),
+        };
```

Comment:
This felt and feels gross to me, but creating a dedicated class with a constructor to tidy it up seemed like overkill.

Reply:
I'm all for typeless, coercive, old school javascript wild west code so I love this kind of stuff, especially now that typescript seems to be the de-facto successor to JS. But, to each their own!